### PR TITLE
refactor(connector): [Rapyd] add and implement the get_currency_unit function

### PR DESCRIPTION
**Pull request by shivansh-bhatnagar18**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

This PR solves issue number #2244 for currency unit conversion in Rapyd Connector

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?
Create a payment via hyperswitch, (Rapyd connector) and match the amount you are passing in your request with amount you are getting in connector response. Both should be same

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/rapyd.rs [hunk 1]
```diff
@@ -483,7 +493,7 @@ impl
         &self,
         req: &types::PaymentsCaptureRouterData,
     ) -> CustomResult<Option<types::RequestBody>, errors::ConnectorError> {
-        let req_obj = rapyd::CaptureRequest::try_from(req)?;
+        let req_obj = rapyd::CaptureRequest::try_from(&req)?;
         let rapyd_req = types::RequestBody::log_and_get_request_body(
```
Comment (SamraatBansal) [inline crates/router/src/connector/rapyd.rs:497]: Hey @shivansh-bhatnagar18 
You have to do the transformation here as well in Capture TryFrom.

You have to do it for `req.request.amount_to_capture`

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

File: crates/router/src/connector/rapyd/transformers.rs [hunk 2]
```diff
@@ -13,6 +13,37 @@ use crate::{
     utils::OptionExt,
 };
 
+#[derive(Debug, Serialize)]
+pub struct RapydRouterData<T> {
+    pub amount: String,
```
Comment (SamraatBansal) [inline crates/router/src/connector/rapyd/transformers.rs:18]: This should be `i64`

File: crates/router/src/connector/rapyd.rs (hunk 1 above)
Comment (shivansh-bhatnagar18) [reply crates/router/src/connector/rapyd.rs:497]: Ohh Thank You for pointing out my mistake. I have made the changes and will be creating a draft PR for the checks.
After these checks I'll create a PR

**Comment by shivansh-bhatnagar18** [comment]
@SamraatBansal Please check out the changes that I have made and point me out if their are any issues

File: crates/router/src/connector/rapyd.rs [hunk 3]
```diff
@@ -64,6 +64,10 @@ impl ConnectorCommon for Rapyd {
         "rapyd"
     }
 
+    fn get_currency_unit(&self) -> api::CurrencyUnit {
+        api::CurrencyUnit::Base
+    }
```
Comment (SamraatBansal) [inline crates/router/src/connector/rapyd.rs:69]: As per Rapyd's Use Case the amount will be `Minor`

```suggestion
    fn get_currency_unit(&self) -> api::CurrencyUnit {
        api::CurrencyUnit::Minor
    }
```

File: crates/router/src/connector/rapyd.rs (hunk 1 above)
Comment (SamraatBansal) [reply crates/router/src/connector/rapyd.rs:503]: This looks good to me, resolving this comment

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

**Comment by SamraatBansal** [comment]
Hey @shivansh-bhatnagar18 I have added some minor comments, otherwise the PR looks good to me, thanks for prompt replies!

File: crates/router/src/connector/rapyd.rs (hunk 3 above)
Comment (shivansh-bhatnagar18) [reply crates/router/src/connector/rapyd.rs:69]: Sorry for that major fault. I don't know why I figured out that it should be base currency from the documentation. I'll do the corrections immediately.

**Comment by shivansh-bhatnagar18** [comment]
@SamraatBansal Thank you for your guidance on this issue. Please Check the changes that I have made.

**Comment by shivansh-bhatnagar18** [comment]
@SamraatBansal  @swangi-kumari Please check the PR for final review

**Comment by shivansh-bhatnagar18** [comment]
@swangi-kumari Sorry for frequent requests but can you re-run the tests?

**Comment by shivansh-bhatnagar18** [comment]
@swangi-kumari Can you please run the final checks. This time the issue is resolved

**Comment by shivansh-bhatnagar18** [comment]
@SamraatBansal @swangi-kumari Can someone help me with tests on the stable toolchain? I am continuously trying to fix the errors thrown by them.

**Review by SamraatBansal — APPROVED** [review]
LGTM

**Comment by shivansh-bhatnagar18** [comment]
@swangi-kumari @SamraatBansal If the Pull Request is good enough, can you help me with the checks and merge the pull request?

**Review by ArjunKarthik — APPROVED** [review]

**Comment by shivansh-bhatnagar18** [comment]
@swangi-kumari Can you re-run the tests ?

**Comment by swangi-kumari** [comment]
Hey @shivansh-bhatnagar18 ,
Thanks a bunch for all your contributions! We’ve got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Pull request merged**